perfdash/local_metrics_dir.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+// LocalMetricsDirectory prepares a client to fetch metrics data from a local directory.
+// This is meant to be used for testing and development to load results from the report-dir output of clusterloader2.
+type LocalMetricsDirectory struct {
+	dirPath string
+}
+
+// NewLocalMetricsDirectory creates a new LocalMetricsDirectory.
+func NewLocalMetricsDir(pathPrefix string) (MetricsBucket, error) {
+	absPath, err := filepath.Abs(pathPrefix)
+	if err != nil {
+		return nil, err
+	}
+
+	return &LocalMetricsDirectory{
+		dirPath: absPath,
+	}, nil
+}
+
+// GetBuildNumbers fetches the build numbers from a local artifacts directory.
+func (b *LocalMetricsDirectory) GetBuildNumbers(job string) ([]int, error) {
+	filePath := b.dirPath + "/" + job
```

Comment:
We should use the platform-specific path separator instead of hard-coding slash. Maybe this doesn't run on Windows anyway, but someone might want it to.